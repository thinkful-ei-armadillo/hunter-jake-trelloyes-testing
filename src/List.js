import React from 'react';
import Card from './Card';

import './List.css';

function List (props) {

  const CardList = props.cards.map((card) => {

    return (
      <Card title={card.title} content={card.content} key={card.key} />
    );
  });

  return (
    <section className="List">
      <header className="List-header">
        <h2>{props.header}</h2>
      </header>
      <div className="List-cards">{CardList}</div>
    </section>
  );
}

export default List;
