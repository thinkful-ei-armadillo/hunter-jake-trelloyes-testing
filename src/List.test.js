import React from 'react';
import List from './List';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {

  const cards = [
    {
      key : 'a',
      title : 'Card Title',
      content : 'Card Content',
    },
  ];

  const div = document.createElement('div');
  ReactDOM.render(<List header cards={cards} />, div);
  ReactDOM.unmountComponentAtNode(div);
});


describe('<List />', () => {
  it('renders the UI as expected', () => {
    const cards = [
      {
        key : 'a',
        title : 'Card Title',
        content : 'Card Content',
      },
    ];
    const tree = renderer
      .create(<List header="example header" cards={cards}  />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });
})