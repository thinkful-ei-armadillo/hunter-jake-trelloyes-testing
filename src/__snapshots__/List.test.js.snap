// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      example header
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        Card Title
      </h3>
      <p>
        Card Content
      </p>
    </div>
  </div>
</section>
`;
